src/main/cli/process.ts

```diff
@@ -0,0 +1,59 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app } from "electron";
+import { ICli, ICliParam } from "readium-desktop/main/cli/commandLine";
+import { Arguments } from "yargs";
+
+declare const __APP_NAME__: string;
+declare const __APP_VERSION__: string;
+
+// Logger
+const debug = debug_("readium-desktop:cli");
+
+export async function processCommandLine(commandLine: ICli[], argv: Arguments): Promise<boolean> {
```

Comment:
Doing this code review, I am now realizing that this implements the layer of logic on top of the low-level `yargs` functionality. Have you considered using `meow` (or similar lib) to easily handle all the edge-cases of options parsing, help display, exit conditions / codes, and perhaps more importantly: Promise "unhandled rejections" which is a likely occurence in our case (lots of await/async without try/catch). See `hard-rejection` in https://github.com/sindresorhus/meow

Reply:
Note that changing the current ad-hoc logic (command line parameter handling) in favour of a more robust mechanism (for example using Meow, but there are others) would require a lot of refactoring and testing. So I am going to resolve this comment for now, and will file a separate issue in due time.